test/test_response.py

```diff
@@ -564,6 +564,50 @@ def test_io_not_autoclose_textiowrapper(self) -> None:
         with pytest.raises(ValueError, match="I/O operation on closed file.?"):
             next(reader)
 
+    def test_read_with_illegal_mix_decode_toggle(self) -> None:
+        compress = zlib.compressobj(6, zlib.DEFLATED, -zlib.MAX_WBITS)
+        data = compress.compress(b"foo")
+        data += compress.flush()
```

Comment:
Please use `data = zlib.compress(b"foo")` instead in this test and the one below.

Reply:
Done. I adapted expectations (read size) to match the initial raw payload struct.